### ISSUE TITLE
Regex error in NodeTaskExecutor

### ISSUE THREAD

**Issue by yhuard** [eirslett/frontend-maven-plugin#192]
Hi

I get the following error when executing goal 'npm':

```
[ERROR] Failed to execute goal com.github.eirslett:frontend-maven-plugin:0.0.23:npm (npm install) on project ****: Execution npm install of goal com.github.eirslett:frontend-maven-plugin:0.0.23:npm failed: Look-behind group does not have an obvious maximum length near index 11
[ERROR] //(?<keep>.*):.*@
```

It is related to the regex in NodeTaskExecutor.java, line 65. I'm using Java 6 and Windows 7 on this project. Could you help me?

**Comment by eirslett** [comment]
Ouch... @kwa could you please have a look at this? I'm not a regex ninja, unfortunately.
@YHuard can you post a simple example file (or project) that reproduces the error?

**Comment by YHuard** [comment]
I tried to run the example projet you give and got the same error. The only thing I changed is the pom.xml with the following configuration:

```xml
                <version>0.0.23</version>

                <executions>

                    <execution>
                        <id>install node and npm</id>
                        <goals>
                            <goal>install-node-and-npm</goal>
                        </goals>
                        <configuration>
                            <nodeVersion>v0.12.2</nodeVersion>
                            <npmVersion>2.7.5</npmVersion>
                        </configuration>
                    </execution>
```

To be more specific, I'm running Maven 3.2.5 (the latest 3.3.x isn't compatible with Java 6) and I use eclipse. I also set a proxy, which is probably why the regex is called

**Comment by eirslett** [comment]
Yup, it's probably because of the proxy. I think you could use version 0.0.22 as well, since you use Maven 3.2. This problem was introduced in version 0.0.23. If you could provide your proxy settings (you can anonymize it of course) that would probably be very helpful!

**Comment by YHuard** [comment]
My proxy is simply a raw ip and a port (`protocol='http', host='17x.xx.xxx.xx', port=8080`), there's no authentication with a password.

Version 0.0.22 works like a charm

**Comment by kwa** [comment]
@YHuard so you have something like this in your settings.xml:
```
<proxies>
  <proxy>
    <id>example-proxy</id>
    <active>true</active>
    <protocol>http</protocol>
    <host>17x.xx.xxx.xx</host>
    <port>8080</port>
  </proxy>
</proxies>
```
?

It would be interesting to know if you get this problem if you run using java 7 or java 8 (I see that you are running in windows so perhaps it is not easy to change...)

On a side note I have been looking into authenticating proxies some more because some people have both "strange" user names and passwords. Strange -> user names like 'l@adder' and passwords like 'a!#:@word', my regex does not take that into account when trying to hide password when it is printed. Also, there 'may' be problems if special characters in the user name and or password are not url-encoded but possibly that is a problem for the npm itself.

Given my assumption about your proxy config it is clear that problem stems from the fact that there is no password to hide, but the regex tries anyway.

I will see what I can come up with and share it.

**Comment by eirslett** [comment]
We might have to remove the password hiding, and suggest users should put their npm proxy configuration inside .npmrc (see https://github.com/eirslett/frontend-maven-plugin/pull/191)
@lillesand do you think it would be okay to remove the proxy functionality from the npm mojo, and replace it with usage of .npmrc? Would it lead to any significant drawbacks? (except duplication of proxy configuration maybe)
Is there anything that cannot be done with .npmrc, that can be done with today's solution?

**Comment by kwa** [comment]
I see the comments in #191 and I do not see .xxxrc files the way to go due to a couple of reasons, but I will post there instead.
I have come up with a solution that handles 'known' assumptions about an URI's  userInfo (as defined by java.net.URI), and skips masking passwords if any of the assumptions fails. Basically everything revolves around identifying a URI with username and password and if it can't be found do nothing.

I guess, however, if each "xxxRunner.java" would implement its own taskToString() they could decide what and how to display the context in which they are running for example their parameters.

**Comment by kwa** [comment]
I have made a pull request #195 .

**Comment by YHuard** [comment]
@kwa that's right, my proxy settings are defined like that:

```xml
  <proxies>
	<proxy>
        <id>proxyid</id>
        <active>true</active>
        <protocol>http</protocol>
        <host>17x.xx.xxx.xx</host>
        <port>8080</port>
    </proxy>
  </proxies>
```
